File: src/main/scala/org/biodatagraphdb/alsdb/value/ProActDemographics.scala
package org.biodatagraphdb.alsdb.value

import org.apache.commons.csv.CSVRecord

case class ProActDemographics(
                             subjectId:Int,
                             subjectGuid:String,
                             age:Int,
                             daysSinceBirth:Int,Sex:String
                             ) {
  val id:String = subjectGuid

}
/* column names
TODO: add support for determining Race attribute
subject_id,Demographics_Delta,Age,Date_of_Birth,
Ethnicity,Race_Americ_Indian_Alaska_Native,Race_Asian,
Race_Black_African_American,Race_Hawaiian_Pacific_Islander,
Race_Unknown,Race_Caucasian,Race_Other,Race_Other_Specify,
Sex
 */
object ProActDemographics extends ValueTrait {
  def parseCSVRecord(record:CSVRecord):ProActDemographics = {
    ProActDemographics(
      record.get("subject_id").toInt,
      generateProActGuid(record.get("subject_id").toInt),
      validIntegerString(record.get("Age")),
      validIntegerString(record.get("Date_of_Birth")),
      record.get("Sex")
    )
  }
}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/ProActDemographics.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

data class ProActDemographics(
        val subjectId: Int,
        val subjectGuid: String,
        val age: Int,
        val daysSinceBirth: Int, val Sex: String
) {
    val id: String = subjectGuid
    /* column names
TODO: add support for determining Race attribute
subject_id,Demographics_Delta,Age,Date_of_Birth,
Ethnicity,Race_Americ_Indian_Alaska_Native,Race_Asian,
Race_Black_African_American,Race_Hawaiian_Pacific_Islander,
Race_Unknown,Race_Caucasian,Race_Other,Race_Other_Specify,
Sex
 */

    companion object : AlsdbModel {
        fun parseCSVRecord(record: CSVRecord): ProActDemographics =
                ProActDemographics(
                        record.get("subject_id").toInt(),
                        generateProActGuid(record.get("subject_id").toInt()),
                        parseValidIntegerFromString(record.get("Age")),
                        parseValidIntegerFromString(record.get("Date_of_Birth")),
                        record.get("Sex")
                )
    }
}